tests/unit/synapseclient/core/upload/unit_test_multipart_upload.py

```diff
@@ -734,90 +892,103 @@ def test_multipart_upload(self):
             (
-                (1234, None, None, False),
+                (1234, None, 0, False),
                 (DEFAULT_PART_SIZE, 1, False)
             ),
 
-            # multiple parts, but less parts than specified max workers
-            # so we expect max workers to be num of parts
-            (
-               (pow(2, 24), pow(2, 23) - 1000, None, False),
-               (pow(2, 23) - 1000, 3, False),
-            ),
-
-            # parts exceeds specified max_threads, so specified max_threads
-            # passes through unchanged, also specify force_restart
+            # specify force_restart
             (
                (pow(2, 28), None, 8, True),
                (DEFAULT_PART_SIZE, 8, True),
             ),
 
-            # many parts, no max_threads, specified, should use default
+            # no max_threads, specified, should use default
             (
                 (pow(2, 28), None, None, False),
                 (DEFAULT_PART_SIZE, pool_provider.DEFAULT_NUM_THREADS, False),
             ),
 
             # part size specified below min, should be raised
             (
-                (1000, 1, None, False),
-                (MIN_PART_SIZE, 1, False),
+                (1000, 1, 5, False),
+                (MIN_PART_SIZE, 5, False),
             ),
 
             # part size would exceed max number of parts,
             # should be adjusted accordingly
             (
-                (pow(2, 36), MIN_PART_SIZE + 1, 32, True),
-                (int(math.ceil(pow(2, 36) / MAX_NUMBER_OF_PARTS)), 32, True),
+                (pow(2, 36), MIN_PART_SIZE + 1, 8, True),
+                (int(math.ceil(pow(2, 36) / MAX_NUMBER_OF_PARTS)), 8, True),
             )
         ]
 
         for (file_size, in_part_size, in_max_threads, in_force_restart),\
             (out_part_size, out_max_threads, out_force_restart)\
                 in tests:
 
+            expected_upload_request = {
+                'concreteType': 'org.sagebionetworks.repo.model.file.MultipartUploadRequest',
+                'contentType': content_type,
+                'contentMD5Hex': md5_hex,
+                'fileName': dest_file_name,
+                'fileSizeBytes': file_size,
+                'generatePreview': True,
+                'storageLocationId': storage_location_id,
+            }
+
             result, upload_mock = self._multipart_upload_test(
                 upload_side_effect,
                 syn,
-                chunk_fn,
-                file_size,
-                in_part_size,
                 dest_file_name,
-                md5_hex,
-                content_type,
-                storage_location_id,
+                in_part_size,
+
+                expected_upload_request,
+                mock.ANY,
+                mock.ANY,
+
                 max_threads=in_max_threads,
                 force_restart=in_force_restart,
             )
 
+            expected_upload_request = {
```

Comment:
this is looks identical to a variable assignment above. is this necessary/intentional?